source/MainWindow.cpp

```diff
@@ -224,6 +224,9 @@ BMessage(B_ABOUT_REQUESTED));
 	fResourceList->AddStatusView(fStatsBox);
 	fStatsBox->AddChild(fStatsString);
 
+	fMergePanel = new BFilePanel(B_OPEN_PANEL, new BMessenger(this), NULL, 0, true,
```

Comment:
BFilePanel constructor will copy the BMessenger and the BMessage. new shouldn't be used.

Reply:
I tried your suggestion, but it doesn't work without "new", because BFilePanel requires pointers in it's constructor.

Reply:
Requiring pointers doesn’t mean requiring `new`. You can pass the address/reference of a `BMessenger` object. If the `BFilePanel` constructor doesn’t take the ownership of the object, the caller is responsible for destroying it.

Reply:
I've now researched addresses and pointers, but I don't find anything that works. I also searched online for uses of BFilePanel, and they all use the same implementation with the new keyword.

The solution I have been able to come up with is initialising the variable and storing it beforehand - (without new), and then pass a reference? Is this what I should do?

Reply:
Yes, and [here](https://github.com/HaikuArchives/Filer/blob/master/sources/AutoFilerTab.cpp#L66) is an example.